### PR TITLE
Adding a few section labels

### PR DESCRIPTION
**Pull request by davidcok**

### CODE REVIEW

File: docs/DafnyRef/UserGuide.md [hunk 1]
```diff
@@ -896,7 +896,7 @@ TO BE WRITTEN
 ### 25.8.4. Java
 
 The Dafny-to-Java compiler writes out the translated files of a file _A_`.dfy`
-to a directory _A_-java. The `-out` option can be used to choose a
+to a directory `_A_-java`. The `-out` option can be used to choose a
```
Comment (atomb) [inline docs/DafnyRef/UserGuide.md:899]:
```suggestion
to a directory _A_`-java`. The `-out` option can be used to choose a
```
Although I could also imagine removing the underscores and including the "A" in the verbatim text, too.

File: docs/DafnyRef/Attributes.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ with no argument is interpreted as if it were true.
 
 ## 23.1. Attributes on top-level declarations
 
-### 23.1.1. `{:autocontracts}` {#sec-attributes-autocontracts}
```
Comment (atomb) [inline docs/DafnyRef/Attributes.md:36]: Why remove this one?

**Review by atomb — CHANGES_REQUESTED** [review]
A very tiny change, and a question.

File: docs/DafnyRef/UserGuide.md (hunk 1 above)
Comment (davidcok) [reply docs/DafnyRef/UserGuide.md:899]: The underscores are italics

File: docs/DafnyRef/Attributes.md (hunk 2 above)
Comment (davidcok) [reply docs/DafnyRef/Attributes.md]: Huh -- must have been an accidental artifact of merging/conflict resolution. I will leave it in.

**Review by atomb — APPROVED** [review]
Looks good!

**Pull request merged**